api/graphql/types/reservation/serializers/confirm_serializers.py

```diff
@@ -91,8 +90,8 @@ def validate(self, data, prefill_from_profile=False):
             payment_type = data.get("payment_type", "").upper()
             reservation_unit = self.instance.reservation_unit.first()
 
-            active_price = ReservationUnitPricingHelper.get_active_price(reservation_unit)
-            if active_price.pricing_type == PricingType.PAID or self.instance.price_net > 0:
+            active_pricing = reservation_unit.actions.get_active_pricing()
+            if active_pricing.highest_price > 0 or self.instance.price_net > 0:
```

Comment:
There are also some more places where the `price_net` can be changed to `price`, but let's maybe do that later since https://github.com/City-of-Helsinki/tilavarauspalvelu-core/pull/1304 is not merged yet.